components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/FetchBuildValidationDataCommand.java

```diff
@@ -132,7 +133,7 @@ private void printFailedRequest(FailedRequestException e) {
         }
     }
 
-    private String buildScanIndexToOrdinal(int i) {
+    private String runNumToOrdinal(int i) {
```

Comment:
I would simplify the implementation to return String.valueOf(i)+ "th" as the default for all values >=4.

You can then also simplify `fetchingMessageFor()` by removing the size check.

Reply:
✅ Done